contrib/tracking_test.go

```diff
@@ -0,0 +1,65 @@
+package contrib
+
+import (
+	"gocv.io/x/gocv"
+	"image"
+	"testing"
+)
+
+func BaseTestTracker(t *testing.T, tracker Tracker, name string) {
+	if tracker == nil {
+		t.Error("TestTracker " + name + " should not be nil")
+	}
+	img := gocv.IMRead("../images/face.jpg", 1)
+	if img.Empty() {
+		t.Error("TestTracker " + name + " input img failed to load")
+	}
+	rect := image.Rect(250, 150, 250+200, 150+250)
+	init := tracker.Init(img, rect)
+	if !init {
+		t.Error("TestTracker " + name + " failed in Init")
+	}
+	_, ok := tracker.Update(img)
+	if !ok {
+		t.Error("TestTracker " + name + " lost object in Update")
+	}
+	img.Close()
+}
+
+func TestTrackerMil(t *testing.T) {
+	tracker := NewTrackerMil()
+	BaseTestTracker(t, tracker, "Mil")
+	tracker.Close()
+}
+func TestTrackerBoosting(t *testing.T) {
+	tracker := NewTrackerBoosting()
+	BaseTestTracker(t, tracker, "Boosting")
+	tracker.Close()
+}
+func TestTrackerMedianFlow(t *testing.T) {
+	tracker := NewTrackerMedianFlow()
+	BaseTestTracker(t, tracker, "MedianFlow")
+	tracker.Close()
+}
+func TestTrackerTld(t *testing.T) {
+	tracker := NewTrackerTld()
+	BaseTestTracker(t, tracker, "Tld")
+	tracker.Close()
+}
+func TestTrackerKcf(t *testing.T) {
+	tracker := NewTrackerKcf()
+	BaseTestTracker(t, tracker, "Kcf")
+	tracker.Close()
+}
+
+func TestTrackerMosse(t *testing.T) {
+	tracker := NewTrackerMosse()
+	BaseTestTracker(t, tracker, "Mosse")
+	tracker.Close()
+}
+
+func TestTrackerCsrt(t *testing.T) {
+	tracker := NewTrackerCsrt()
+	BaseTestTracker(t, tracker, "Csrt")
+	tracker.Close()
+}
```

Comment:
Instead of creating many test functions, we could use a [table-driven test](https://github.com/golang/go/wiki/TableDrivenTests), WDYT?

Reply:
I really like the DRY approach BTW

Reply:
good idea, a table of functions was a bit over my head, though. will have to look it up.

Reply:
@berak try something like this:

```go
func TestTrackers(t *testing.T) {
	tab := []struct {
		name    string
		tracker Tracker
	}{
		{"Mil", NewTrackerMil()},
		{"Boosting", NewTrackerBoosting()},
	}

	for _, tt := range tab {
		if tt.tracker == nil {
			t.Error("TestTracker " + tt.name + " should not be nil")
		}

		img := gocv.IMRead("../images/face.jpg", 1)
		defer img.Close()

		if img.Empty() {
			t.Error("TestTracker " + tt.name + " input img failed to load")
		}

		rect := image.Rect(250, 150, 250+200, 150+250)

		if init := tt.tracker.Init(img, rect); !init {
			t.Error("TestTracker " + tt.name + " failed in Init")
		}

		_, ok := tt.tracker.Update(img)
		if !ok {
			t.Error("TestTracker " + tt.name + " lost object in Update")
		}
	}
}
```

Reply:
hehe, that's what i'm doing now ;)

(and it revealed the problem with Close, since all we have is a Tracker now)